### ISSUE TITLE
httpsServer setting clear on boot

### ISSUE THREAD

**Issue by Tom-Jans** [pimatic/pimatic#417]
Im trying to acces my pimatic trough https but when i add the httpsServer settings from http://pimatic.org/guide/getting-started/remote-ssl/ they get removed on boot.

I add this to the config
```
"httpsServer": {
  "enabled": true,
  "port": 443,
  "hostname": "",
  "keyFile": "ca/pimatic-ssl/private/privkey.pem",
  "certFile": "ca/pimatic-ssl/public/cert.pem",
  "rootCertFile": "ca/certs/cacert.crt"
},
```
and after boot it changes into this
"httpsServer": {},

**Comment by sweetpi** [comment]
@Tom-Jans Does the problem still exist? Does it only happen on pi boot or on each restart of pimatic?

I can't find the source of the probem. Could you post or email me (see bottom of README) your config?

**Comment by Heizelmann** [comment]
I had the same problem. Check if you have a duplicate entry for https at the end of the confg section.

As a novice user I started with the default config file as template. In this file only an httpServer and no httpsServer is defined. After defining and saving some rules and variables via the web user interface the config file is stored back by pimatic. After the first write pimatic adds an empty https entry at the  end of the settings part. Because it is at the end and not by the http entry  at the top this can easily be overseen. 
So I inserted a new https entry as described in http://pimatic.org/guide/getting-started/remote-ssl/ right after the http entry near the top of the settings part. So accidently I had two https entries in settings:  the fully qualified at the top and the empty entry at the end of the settings part. While this is still a valid JSON the JSON parser overrides the first setting with the second on the next read and removes the first one on the next write.
